kubeutils/annotations.go

```diff
@@ -0,0 +1,29 @@
+package kubeutils
+
+// SoloClusterAnnotation was originally implemented in solo-kitv2's ezkube package
+// It allows for easy mimicking of the old clustername field present in
+// k8s prior to 1.24. It is stored here to make future changes easier and not
+// require multi package switches as if any of the dependant packages are out of
+// sync on this field it can cause hard to diagnose issues.
+const SoloClusterAnnotation = "cluster.solo.io/cluster"
+
+type hasAnnotations interface {
```

Comment:
nit: interface naming - hasAnnotation sounds more like a function. what do you think of something like ObjectWithAnnotations? also do we need to export it for other repos to use?

Reply:
This interface is used purely to not have to care about the outside interfaces so id rather not export it so it is implicitly local